WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentService.swift

```diff
@@ -66,7 +66,9 @@ final class CardPresentPaymentService: CardPresentPaymentFacade {
         case .completed(let cardReader, _):
             let connectedReader = CardPresentPaymentCardReader(name: cardReader.name ?? cardReader.id,
                                                                batteryLevel: cardReader.batteryLevel)
-            paymentEventSubject.send(.idle)
+            paymentEventSubject.send(.show(eventDetails: .connectionSuccess(done: { [weak self] in
+                self?.paymentEventSubject.send(.idle)
+            })))
```

Comment:
Not for this PR or this change, but question: We currently do not have tests in place for the `CardPresentPaymentService`, right? (I'm seeing usage of `MockCardPresentPaymentService` as part of the tests for totals, but I think that's it)

Reply:
Nope. The service is as thin a wrapper as it can be around the existing code, so it doesn't make a lot of sense to write tests for it. 

Of course, it would be nice if all the code it wrapped was fully tested (it isn't) but I don't think that changes the calculation right now.

If/when we replace the underlying code, we should test it as part of that (massive) project, we should add tests to the service then, because it'll have responsibilities beyond adapting the calls.